rfcs/reftest_simplification.md

```diff
@@ -0,0 +1,56 @@
+# RFC #15: Reftest graph simplification
+
+## Summary
+
+Simplify what we allow within the reftest graph.
+
+## Details
+
+### Background
+
+Currently, reftests form a graph which can quickly end up with a large number of pass conditions.
+
+In rough principle, we load everything with `<link rel=match>`/`<link rel=mismatch>` into a directed graph (with URLs as vertices, `match`/`mismatch` as edges), and then make every source node (i.e., a node with no incoming edges) into a "test".
+
+From each test, we enumerate all possible walks from the source node with either no repeating nodes and ending at a sink node (i.e., a node with no outgoing edges) or a ending at the first repeated node.
+
+In reality, we have very little complexity within the graph, as of [37e794f690](https://github.com/web-platform-tests/wpt/tree/37e794f69091ffed1dfa7399a3ef154125fddda9): we have 14954 tests, and only 219 are more complex than a simple two page comparison.
+
+Of those 219:
+ * 111 just have alternates at the top level (the basic OR case),
+ * 55 are trails with no repeating vertices (the basic AND case),
+ * 48 are trails ending in a cycle (still a basic AND),
+ * leaving us with 5 more complex cases (which mix AND and OR).
+ 
+These five more complex cases are given [here](https://gist.github.com/gsnedders/7874fe11acd1dc8eaacb7448db8ca690). All but one of these are `css-transforms` that have been known to be broken for years and are quite clearly bogus (many pass conditions are subsets of one another!). The only other example is `/css/CSS2/text/text-indent-wrap-001.xht`, which must be \[(not) equal to `/css/CSS2/text/text-indent-wrap-001-notref-block-margin.xht`] OR \[equal to `/css/CSS2/text/text-indent-wrap-001-ref-inline-margin.xht` AND `/css/CSS2/text/text-indent-wrap-001-ref-float.xht`]. Looking at the individual tests, it appears that all of these are mistaking multiple links as conjunctions, and all of these should be simple chains rahter than alternates; [#15523](https://github.com/web-platform-tests/wpt/pull/15523) does this. (This also makes me suspicious as to how many of the 111 tests we have with alternates make a similar mistake, and whether we can make this clearer.)
```

Comment:
Which mistake?

The `css-transforms` mistakes? They're impossible after this RFC given we'd never have AND/OR combined like that. Otherwise, we could, but we'd need to compute the reftest graph for lint which we've thus far avoided doing (and will introduce even further differences between `./wpt lint --all` and `./wpt lint` given we don't deal with dependencies at all).

The 111 tests? I don't think I can teach a machine to judge authorial intention. 🙂

Reply:
Sounds like we don't lint at all for reftest graph structure, which we could, but let's treat that as separate from this RFC.